### DISCUSSION TITLE
The links to config files in /config/Helper-Script is read only. The fans can not be activated from the mainsail menu

### DISCUSSION THREAD

**Discussion by ACKET2009** [Guilouz/Creality-Helper-Script-Wiki#548] [Issues]
Hello! I was trying to edit some macros in config files from the mainsail config menu, but can not save any changes...
And additional - I can't activate the fans from the mainsail desktop.
Yesterday I has installed the last version of firmware to the K1max printer, and make a full reset to factory settings, then install the script.
Now trying to set the nice print as it was before, but have this troubles...
P.S. Additionally, during the print, the layers always show the 0 of nmbr of layers - only the height by Z is correct information to check the process...

**Comment by Guilouz** [comment]
All config files are symlinked to keep updated when script is updated. They are not supposed to be modified

Issue with fans was an issue with Mainsail that was fixed in last version of Mainsail.

For layers depending slicer you use.

**Comment by ACKET2009** [comment]
Hm... I'm using the latest version of the mainsail 2.11.1... The slicer is OrcaSlicer 1.8.0 beta2
Before the upgrade and installing new version of script the coolers works from the mainsail panel...

**Reply by Guilouz** [reply]
Nothing to see with script, it's an issue with output_pin and Mainsail that I have reported here: https://github.com/mainsail-crew/mainsail/issues/1869.
Latest version is 2.11.2 with this fix: https://github.com/mainsail-crew/mainsail/releases/tag/v2.11.2

**Reply by Guilouz** [reply]
This is what you need to have in your gcodes on orca:

<img width="1094" alt="Capture d’écran 2024-05-04 à 23 36 03" src="https://github.com/Guilouz/Creality-Helper-Script-Wiki/assets/12702322/a165e696-4db3-45f7-8221-3b393bf61254">

**Reply by ACKET2009** [reply]
Thank's! (I do not write the string with layer change SET_PRINT_STATS_INFO CURRENT_LAYER={layer_num + 1} in my GCODE) 
About mainsail - Can It be upgraded from the mainsail shell? Now it shows no updates, that the latest version is already installed...

**Comment by Guilouz** [comment]
Use the refresh button in Update Manager to check updates

**Comment by ACKET2009** [comment]
Thank You very much! The problem solved!
